### ISSUE TITLE
Cap model sometimes loses calibration point and inverts

### ISSUE THREAD

**Issue by MathijsG** [opentrack/opentrack#289]
I recently soldered a Cap model, so I could test both types of optical tracking, clip and cap. I like most of the parts more of the cap: the axis are less interconnected, so when you zoom you don't also pitch up, and you can make nice rolls without moving to the left or right. 

However I also noticed some downsides, which I don't know are maybe something in the software or an inherent problem of the design:

When I sometimes look away from the screen, the red cursor is moving to a different position and the orientation is totally messed up. Recentering doesn't work then, only actually stopping the tracking again. I do like the recent keymap options for toggling the tracking itself, because they could come in handy when this happens.

Maybe it's a bit hard to understand, but here are some pictures:

Before:

![image](https://cloud.githubusercontent.com/assets/529552/12098406/7ab49a60-b322-11e5-9137-ea19101e1918.png)

After:

![image](https://cloud.githubusercontent.com/assets/529552/12098445/a7b2188a-b322-11e5-851a-a671c419ffaa.png)

It acts somewhat inverted somehow.

I just discovered that I also can reset its behavior again by moving too far up so the two bottom leds will be above the top led, and when I then move them down again they are normal.

Will make a video of it when I have the time.

**Comment by sthalik** [comment]
Do you have dynamic pose resolution set, and its timer set to at least a second or so?

**Comment by MathijsG** [comment]
![image](https://cloud.githubusercontent.com/assets/529552/12175845/bae80d8a-b564-11e5-9b8c-c68dee9f318e.png)


I should set it to 1000MS? Will try. What exactly does it do anyway? Couldn't really figure out when I googled it earlier.

**Comment by MathijsG** [comment]
I still have with 1000MS that when I shake wildly, or after a while the octopus isn't really rotating around the middle anymore, but it looks like it is going in a circle around. Should make a video of it :-P.

**Comment by sthalik** [comment]
@MathijsG "dynamic pose" helps remember which point is which. With it off, no information is preserved.

Look for any anomalies like a point not visible after rotating a lot, or points merging.

**Comment by MathijsG** [comment]
Will check further. Why not keep this to for example 10.000MS? Does that cost more CPU or RAM or something?

**Comment by sniperwolfpk5** [comment]
Is it better to turn on dynamic pose? any benifits using this option.

**Comment by sthalik** [comment]
@MathijsG it's meant not to reset on brief failure

@sniperwolfpk5 yes

**Comment by sthalik** [comment]
> Does that cost more CPU or RAM or something?

It doesn't cost any more cpu time or memory no matter the setting.

If you set it to a really high value it may remember an incorrect position for too long. Set it to a typical amount your LEDs may be not recognizable, with a margin of error.

**Comment by MathijsG** [comment]
Could it be that this is fixed in the latest build (P10)? How long I look behind my doesn't matter anymore, even when I look behind me 10+ seconds, when I look back the orientation is normal, instead of weird.

THANKS! :-)

**Comment by sthalik** [comment]
@MathijsG this isn't a bug. Dynamic pose timeout happens and the software can no longer distinguish points from memory. Return to center pose and cover one of the LEDs, or toggle tracking twice.

**Comment by MathijsG** [comment]
But when I set the Dynamic Pose timeout for example to 9999 MS (9,9 seconds). The program already loses all orientation while looking away for example for just 1,5 second. This is pretty annoying when you are looking away for a brief moment.

**Comment by sthalik** [comment]
It remembers exact on-screen positions and they don't unambiguously correspond to LED lights when your model return to the view.

**Comment by MathijsG** [comment]
But I think it's very common to be able to look away from the screen for a brief moment and resume looking. With a clip this is no problem here.

When I disable Dynamic Pose Resolution I have the problem that my yaw range is very limited, when the top led is vertically aligned with one of the outer leds it stops moving, while with Dynamic Pose enabled it can go further. Will make a movie if this explanation is too hard to follow :-P.

**Comment by sthalik** [comment]
I understand your explanation. But there seems to be no good way to take the problem you're describing into account.

**Comment by MathijsG** [comment]
Okay, I understand. But I don't know why it sometimes flips back normally on certain settings. I will still test further. Don't know if other people with a Cap like the official trackhat have this problem?

**Comment by sthalik** [comment]
What flips back again?

**Comment by MathijsG** [comment]
Sorry I mean when I get back in the webcam detection area, most of the times the octopus is being backwards and everything is inverted, but it sometimes DOES work normally. That's why I earlier thought this issue was fixed.

**Comment by sthalik** [comment]
Dynamic pose remembers the last point positions where they were visible. If two points (after they become visible again) is closest to a single one of the remembered positions, dynamic pose gets reset.

It may also give a bad result if wrong points are closest to the remembered points.

**Comment by MathijsG** [comment]
Yeah I think that is the problem somewhere. A clip is mostly 3 points vertically aligned, which can't really be mistaken for another pose, while a cap is some triangle form, and when you turn too far while moving out of the screen, the top led can be mistaken for a led from one of the sides, rendering the octopus sideways.

**Comment by sthalik** [comment]
I don't see any technical solution here. You can reopen with a technical solution.

**Comment by MathijsG** [comment]
Update: When I set the dynamic pose time out to extreme low values, like 1 ms it's flipping back normally again after going back in screen.

**Comment by sthalik** [comment]
@MathijsG the build I linked in the other issue has a "reset tracker" shortcut. There's no better solution.

Effectively disabling dynamic pose is a bad idea.

**Comment by sthalik** [comment]
You can cover a single LED for the timeout's duration and it has the same effect as the reset function.
